dnf-behave-tests/features/repoquery/weak-deps.feature

```diff
@@ -0,0 +1,402 @@
+Feature: Tests for repoquery weak deps related functionality:
+ --recommends, --supplements, --suggests, --enhances,
+ --whatrecommends, --whatsupplements, --whatsuggests, --whatenhances
+ --whatdepends
+
+# all the packages a1, a2, b1, b2, c1, c2, d1, d2 and top follow
```

Comment:
What about to rename package a1 to pkgA1? They could provide provideA1. Or change it to some real name or describe in the name the purpose of the package?

Reply:
@m-blaha, @pkratoch what do you think of the names of packages in this file (and possibly the other ones) and do we have any conventions I should follow?

I simply have a bunch of packages here for which I want to test some non-trivial rich dependencies, so I'm not sure anything more fancy makes sense.

IMO adding a prefix like `pkgA1` doesn't add value and just makes lines longer. The provides are always named like `PKGNAME-provX`, so they are differentiated.

Reply:
I don't think that "pkg" prefix adds any value, but differentiating between ``a``, ``b``, ``c`` and ``d`` would be nice if possible, because otherwise the reader must remember what's the difference between them. I'm thinking of something like ``recommender``, ``supplementer``, ``suggester`` and ``enhancer``, because the packages in the ``repoquery-weak-deps`` repo are this simple (but it's just a suggestion, the schema was helpful enough once I read the specs).

Btw, you are mentioning rich dependencies in your comment, but these are normal ones (rich are the boolean ones, right?). Is this a mistake? Or are you using packages from the ``repoquery-weak-deps`` repo in some other tests, too?

Reply:
As I said, with the schema, the tests were readable enough for me.

> I'm thinking of something like `recommender`, `supplementer`, `suggester` and `enhancer`.

I am not super happy about it either, in other tests it works better (in `dnf-ci-obsoletes`, there is `PackageA` and `PackageA-Obsoleter` and it's clear what are the roles), but here both sides of the relations are important and only one is described by the name, so it's not that great.